simulariumio/data_objects/agent_data.py

```diff
@@ -429,12 +429,19 @@ def from_dimensions(
             ),
         )
 
+    def total_steps(self) -> int:
+        """
+        Get number of frames
+        Use n_timesteps to limit times if it has been provided
+        """
+        return self.n_timesteps if self.n_timesteps >= 0 else len(self.times)
+
```

Comment:
this is convenience for `BinaryWriter`

Reply:
as a reader of this code, I think it would help if you had consistent naming: choose one of `timesteps`, `steps` or `frames` if possible.  All 3 are used in the above function and it's confusing.